### PR TITLE
rivet: apply patches to fix several issues with make-plot

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116016` at 9684f2b5 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>herwig</li>
    <li>sherpa</li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/3a138fef6fae07fb067e31062d251fe9">rivet</a></li> (python38Packages.rivet)</li>
    <li><a href="https://gist.github.com/80293420ac80f1e621bfa869e61cd395">python39Packages.rivet</a></li></li>
    <li><a href="https://gist.github.com/6039f09c5d7cd9932ba81128f0ba0cbe">thepeg</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/libraries/physics/rivet/default.nix:13:5:
```
   |
13 |     ./darwin.patch # configure relies on impure sw_vers to -Dunix
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/physics/rivet/default.nix:101:5:
```
    |
101 |     license     = licenses.gpl3;
    |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `gsl`.
Near pkgs/development/libraries/physics/rivet/default.nix:1:77:
```
  |
1 | { lib, stdenv, fetchurl, fetchpatch, fastjet, fastjet-contrib, ghostscript, gsl, hepmc, imagemagick, less, python3, rsync, texlive, yoda, which, makeWrapper }:
  |                                                                             ^
```
</li>
  </ul>
</details>

File: pkgs/development/libraries/physics/rivet/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchurl, fastjet, fastjet-contrib, ghostscript, gsl, hepmc, imagemagick, less, python3, rsync, texlive, yoda, which, makeWrapper }:
+{ lib, stdenv, fetchurl, fetchpatch, fastjet, fastjet-contrib, ghostscript, gsl, hepmc, imagemagick, less, python3, rsync, texlive, yoda, which, makeWrapper }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/physics/rivet/default.nix:1]:
```suggestion
{ lib, stdenv, fetchurl, fetchpatch, fastjet, fastjet-contrib, ghostscript, hepmc, imagemagick, less, python3, rsync, texlive, yoda, which, makeWrapper }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 116016` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>herwig</li>
    <li>rivet (python38Packages.rivet)</li>
    <li>python39Packages.rivet</li>
    <li>sherpa</li>
    <li>thepeg</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@veprbl fails to build on ofborg

**Pull request merged**